x/harvest/keeper/borrow.go

```diff
@@ -0,0 +1,138 @@
+package keeper
+
+import (
+	"math"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/kava-labs/kava/x/harvest/types"
+)
+
+// USDX is the USDX coin's denom
+const USDX = "usdx"
+
+// Borrow funds
+func (k Keeper) Borrow(ctx sdk.Context, borrower sdk.AccAddress, amount sdk.Coin) error {
+
+	err := k.ValidateBorrow(ctx, borrower, amount)
+	if err != nil {
+		return err
+	}
+
+	err = k.supplyKeeper.SendCoinsFromAccountToModule(ctx, borrower, types.ModuleAccountName, sdk.NewCoins(amount))
+	if err != nil {
+		return err
+	}
+
+	borrow, found := k.GetBorrow(ctx, borrower, amount.Denom)
+	if !found {
+		borrow = types.NewBorrow(borrower, amount)
+	} else {
+		borrow.Amount = borrow.Amount.Add(amount)
+	}
+
+	k.SetBorrow(ctx, borrow)
+
+	ctx.EventManager().EmitEvent(
+		sdk.NewEvent(
+			types.EventTypeHarvestBorrow,
+			sdk.NewAttribute(sdk.AttributeKeyAmount, amount.String()),
+			sdk.NewAttribute(types.AttributeKeyBorrower, borrow.Borrower.String()),
+			sdk.NewAttribute(types.AttributeKeyBorrowDenom, borrow.Amount.Denom),
+		),
+	)
+
+	return nil
+}
+
+// ValidateBorrow validates a borrow request against borrower and protocol requirements
+func (k Keeper) ValidateBorrow(ctx sdk.Context, borrower sdk.AccAddress, amount sdk.Coin) error {
+	if err := k.validateBorrowUser(ctx, borrower, amount); err != nil {
+		return err
+	}
+
+	if err := k.validateBorrowSystem(ctx, borrower, amount); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (k Keeper) validateBorrowUser(ctx sdk.Context, borrower sdk.AccAddress, amount sdk.Coin) error {
```

Comment:
add a comment that describes what this does - I would change name as 'BorrowUser' doesn't seem descriptive. Maybe `validateLoanToValue`?

Reply:
Refactored this function to `ValidateBorrow` in https://github.com/Kava-Labs/kava/pull/702/commits/277c4b1936981182fc09ae0c6ea0973d9cbbca8a.

Also moved price fetching to a helper function. It's a bit cleaner, but still not convinced it's the best way as we're still double-fetching the borrow asset's MoneyMarket.